src/main/java/sirius/db/mixing/SmartQuery.java

```diff
@@ -579,23 +579,31 @@ private void appendFieldList(Compiler c, boolean applyAliases) {
             Tuple<String, EntityDescriptor> joinInfo = c.determineAlias(field.getParent());
             c.getSELECTBuilder().append(joinInfo.getFirst());
             c.getSELECTBuilder().append(".");
-            if (Entity.ID.getName().equals(field.getName()) || Entity.VERSION.getName().equals(field.getName())) {
-                c.getSELECTBuilder().append(field.getName());
-            } else {
-                c.getSELECTBuilder().append(joinInfo.getSecond().getProperty(field.getName()).getColumnName());
-            }
+            String columnName = fetchEffectiveColumnName(field, joinInfo);
+            c.getSELECTBuilder().append(columnName);
+
             if (!c.defaultAlias.equals(joinInfo.getFirst())) {
                 if (applyAliases) {
                     c.getSELECTBuilder().append(" AS ");
                     c.getSELECTBuilder().append(joinInfo.getFirst());
                     c.getSELECTBuilder().append("_");
-                    c.getSELECTBuilder().append(field.getName());
+                    c.getSELECTBuilder().append(columnName);
                 }
                 c.createJoinFetch(field);
             }
         }
     }
 
+    private String fetchEffectiveColumnName(Column field, Tuple<String, EntityDescriptor> joinInfo) {
+        String columnName = null;
```

Comment:
Remove this useless assignment to local variable "columnName"


> A dead store happens when a local variable is assigned a value, including null, that is not read by any subsequent instruction. Calculating or retrieving a value only to then overwrite it or throw it away, could indicate a serious error in the code. Even if it's not an error, it is at best a waste of resources.
Even assigning null to a variable is a dead store if the variable is not subsequently used. Assigning null as a hint to the garbage collector used to be common practice, but is no longer needed and such code should be eliminated.